Add Troubleshooting section about antivirus false positive (#1840)
brimdata/brim: docs/Troubleshooting-(v0.25.0+).md

from experienced members of the Brim community on ways we can best address
the topic. Please come talk to us on [public Slack](https://www.brimsecurity.com/join-slack/)
if you feel you could offer further assistance.

It may be helpful to first understand the role of this binary. As the name
suggests, the binary is part of the [Suricata threat detection engine](https://suricata.io/)
that's bundled with Brim as part of the [Brimcap](https://github.com/brimdata/brimcap)
tool that generates summary logs from packet captures. When used in a default
configuration, Brim invokes Suricata's update utility each time the app
launches to ensure the most recent threat rules are locally cached such that
accurate alerts can be generated from pcaps you import into Brim.

Considering how the tool functions and how Brim uses it, it becomes plausible
that a conservative antivirus engine may perceive the observed network
behavior as consistent with that of ["backdoor"](https://en.wikipedia.org/wiki/Backdoor_(computing))
malware that may reach out to arbitrary network destinations in the name of
transmitting sensitive information or providing network channels for

> nefarious remote logins. Of course, the [functionality of Suricata-Update](https://suricata.readthedocs.io/en/suricata-6.0.0/rule-management/suricata-update.html)
> as described by the OISF is of course quite benign: In its default

Reviewer: ```suggestion
nefarious remote logins. Of course, the [functionality of Suricata-Update](https://suricata.readthedocs.io/en/suricata-6.0.0/rule-management/suricata-update.html)
as described by the OISF is quite benign: In its default
```

> configuration as it's used by Brimcap/Brim, the network destinations it

Reviewer: ```suggestion
configuration as it's used by Brimcap and Brim, the network destinations it
```

bundled with each platform and hence run without external dependencies.

If you wish to perform your own audit to confirm that the executable has not
been tampered with in the build process, below are some pointers to trace
its origin.

* A Brim-managed [fork of suricata-update](https://github.com/brimdata/suricata-update) holds a
  copy of the [original OISF repo](https://github.com/OISF/suricata-update).
  As of Brim
  release `v0.25.0`, the fork is holding a copy of the OISF repo as of tag
  `1.2.0`.

* A branch [`fix-windows`](https://github.com/brimdata/suricata-update/tree/fix-windows)
  exists in the Brim-managed repo that includes minimal changes that we found
  necessary for the executable to build and run on Windows. The total set of
  changes can be reviewed [here](https://github.com/brimdata/suricata-update/compare/1.2.0...0b10a83).

* A [build-suricata repo](https://github.com/brimdata/build-suricata) contains automation that's
  used to create the Suricata packages that are bundled with Brimcap. This
  includes [script code](https://github.com/brimdata/build-suricata/blob/9847f9ba72b24a132aa7068a15b9d3ff9e63af1b/.github/workflows/windows-build.yaml#L103-L109)

>   that unpacks the enhanced Python and runs PyInstaller to generate the

Reviewer: This reads as though we've modified Python rather than Suracata-Update.
```suggestion
  that unpacks the enhanced Suricata-Update and runs PyInstaller to generate the
```
